rotkehlchen/accounting/structures/base.py

```diff
@@ -277,6 +279,24 @@ def get_identifier(self) -> str:
     def get_assets(self) -> list[Asset]:
         return [self.asset]
 
+    def has_details(self) -> bool:
+        return (
+            self.extra_data is not None and
+            ('sub-swaps' in self.extra_data or 'liquity-staking' in self.extra_data)
+        )
+
+    def get_details(self) -> Optional[dict[str, Any]]:
+        if self.extra_data is None:
+            return None
+
+        details = {}
+        if 'sub-swaps' in self.extra_data:
+            details['sub-swaps'] = self.extra_data['sub-swaps']
+        if 'liquity-staking' in self.extra_data:
+            details['liquity-staking'] = self.extra_data['liquity-staking']
```

Comment:
this can be an `elif` right? a single event belongs only to one protocol

Reply:
Turned into a loop